fdbserver/DeltaTree.h

```diff
@@ -391,7 +391,8 @@ struct DeltaTree {
 			DecodedNode* n = root;
 			bool addLeftChild = false;
 
-			while (n != nullptr) {
+			ASSERT(n != nullptr);
```

Comment:
nit: this assert could also go into the loop (though if the pointer never changes, this might not be necessary

Reply:
`n` changes in the loop in two places, and in both places it is set to pointers that were just checked for `nullptr` and if they are `nullptr` then a `break` is done.

So the assert is not necessary at all.

Reply:
Ideally an assert is never necessary :)

Reply:
I guess if we wanted to super precise about the intent we should probably just assert right before the dereference

Reply:
Yes, that would be the correct place to check, and it would be encountered about 7x less frequently.  While insert is not a particularly hot path in DeltaTree, at least compared to things like the seek methods which can be hit 2 to 3 million times per second, I do try to keep it fairly streamlined.

The larger context here is that `n` can only be null if the root is null, which can only happen if the DeltaTree is empty, and insertions are never done on empty DeltaTrees.

Reply:
Can we move the assert to just before this?
```
			const T* prev = addLeftChild ? n->prev : &n->item;
			const T* next = addLeftChild ? &n->item : n->next;
```

Reply:
Done